fitwsarah_frontend/src/components/clientProfile/setting.js

```diff
@@ -2,56 +2,171 @@
+        // Implement your logic to update user information here
+        console.log('Update clicked');
+        console.log('New username:', username);
+        console.log('New first name:', firstName);
+        console.log('New last name:', lastName);
+        console.log('New email:', email);
+    };
 
     useEffect(() => {
-        const fetchProfile = async () => {
-            if (isAuthenticated) {
-                try {
-                    const accessToken = await getAccessTokenSilently();
-                    const response = await fetch('http://localhost:8080/api/v1/accounts/uuid-acc2', {
-                        method: 'GET',
-                        headers: {
-                            Authorization: `Bearer ${accessToken}`,
-                            'Content-Type': 'application/json',
-                        },
-                    });
-
-                    if (!response.ok) {
-                        throw new Error(`Network response was not ok ${response.statusText}`);
-                    }
-
-                    const data = await response.json();
-                    setProfile(data);
-                } catch (error) {
-                    console.error('Error fetching account details:', error);
-                }
-            }
+        if (user && user.picture) {
+            setProfilePicUrl(user.picture);
+        }
+    }, [user]);
+
+    const getAccessToken = useGetAccessToken();
+
+    useEffect(() => {
+        const fetchToken = async () => {
+            const token = await getAccessToken();
+            if (token) setAccessToken(token);
         };
 
-        fetchProfile();
-    }, [getAccessTokenSilently, isAuthenticated]);
+        fetchToken();
+    }, [getAccessToken]);
+
+    useEffect(() => {
+        if (user && user.sub) {
+            getAccountByUserId(user.sub);
+        }
+        if (accessToken) {
+            getAppointmentsByAccountId("dc2b4f0f-76da-4d1e-ad2d-cebf950e5fa2");
+        }
+    }, [user]);
+
+
+    const getAccountByUserId = (userId) => {
+        fetch(`http://localhost:8080/api/v1/accounts/${userId.replace("|", "%7c")}`, {
+            method: "GET",
+            headers: new Headers({
+                "Authorization": `Bearer ${accessToken}`,
+                "Content-Type": "application/json"
+            })
+        })
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error('Network response was not ok ' + response.statusText);
+                }
+                return response.json();
+            })
+            .then((data) => {
+                setProfile(data);
+                console.log(data)
+            })
+            .catch((error) => {
+                console.error("Error fetching service details for userId", userId, ":", error);
+            });
+    };
+
+    const getAppointmentsByAccountId = (accountId) => {
+        fetch(`http://localhost:8080/api/v1/appointments/account/${accountId}`, {
```

Comment:
Is this using auth0 id or the id located in the database?

Reply:
Currently, since the booking appointment by id is not completed right now its using the UUID, user id uses Auth ID but I can't implement it yet because the the booking appointment is not done.